lisa/sut_orchestrator/azure/features.py

```diff
@@ -2151,6 +2162,135 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
+        )
+
+        node_availability_capabilities: List[AvailabilitySettings] = list(
+            environment.nodes.list_features_by_type(AvailabilitySettings)
+        )
+        if node_availability_capabilities and not is_maximize_capability:
+            # Get min capability of all the nodes
+            cap_intersection = node_availability_capabilities[0]
+            for cap in node_availability_capabilities:
+                cap_intersection.intersect(cap)
+
+            assert isinstance(cap_intersection.availability_type, search_space.SetSpace)
+
+            if is_ultra_disk:
+                cap_intersection.availability_type.discard(
+                    AvailabilityType.AvailabilitySet
+                )
+                if (
+                    AvailabilityType.AvailabilityZone
+                    in cap_intersection.availability_type
+                ):
+                    cap_intersection.availability_type.discard(
+                        AvailabilityType.NoRedundancy
+                    )
+
+            # Set ARM parameters based on min capability
+            if params.availability_type == AvailabilityType.Default:
+                params.availability_type = (
+                    cap_intersection._resolve_availability_type_by_priority(
+                        params
+                    ).value
+                )
+            if (
+                params.availability_zones
+                and params.availability_type == AvailabilityType.AvailabilityZone
+            ):
+                params.availability_zones = [
+                    zone
+                    for zone in params.availability_zones
+                    if zone in cap_intersection.availability_zones
+                ]
+                assert params.availability_zones, (
+                    "Invalid zones provided. "
+                    "This SKU in this location supports zones: "
+                    f"{cap_intersection.availability_zones}. "
+                )
+            elif cap_intersection.availability_zones:
+                params.availability_zones = [
+                    cap_intersection.availability_zones.items[0]
+                ]
+
+            assert params.availability_type in [
+                type.value for type in AvailabilityType
+            ], ("Not a valid Availability Type: " f"{params.availability_type}")
+
+            assert (
+                AvailabilityType(params.availability_type)
+                in cap_intersection.availability_type
+            ), (
+                f"Availability Type "
+                f"'{params.availability_type}' "
+                "is not supported in the current configuration. Please select one of "
+                f"{[type.value for type in cap_intersection.availability_type.items]}. "
+                "Or consider changing the disk type or location."
+            )
+
+        if params.availability_type == AvailabilityType.AvailabilitySet:
+            params.availability_zones.clear()
```

Comment:
Add comments like "availability set conflicts with availability zones, and the availability set is higher priority than zones"...